File: app/src/main/java/im/tox/antox/callbacks/AntoxOnFileSendRequestCallback.java
package im.tox.antox.callbacks;

import android.content.Context;
import android.util.Log;

import im.tox.antox.tox.ToxSingleton;
import im.tox.antox.utils.AntoxFriend;
import im.tox.jtoxcore.callbacks.OnFileSendRequestCallback;
import im.tox.jtoxcore.callbacks.OnTypingChangeCallback;

public class AntoxOnFileSendRequestCallback implements OnFileSendRequestCallback<AntoxFriend> {

    private static final String TAG = "OnFileSendRequestCallback";
    private Context ctx;
    ToxSingleton toxSingleton = ToxSingleton.getInstance();

    public AntoxOnFileSendRequestCallback(Context ctx) { this.ctx = ctx; };

    public void execute(AntoxFriend friend, int filenumber, long filesize, byte[] filename) {
        Log.d(TAG, "execute");
    }
}

File: app/src/main/scala/im/tox/antox/callbacks/AntoxOnFileSendRequestCallback.scala
package im.tox.antox.callbacks

import android.content.Context
import android.util.Log
import im.tox.antox.callbacks.AntoxOnFileSendRequestCallback._
import im.tox.antox.tox.ToxSingleton
import im.tox.antox.utils.AntoxFriend
//remove if not needed

object AntoxOnFileSendRequestCallback {

  private val TAG = "OnFileSendRequestCallback"
}

class AntoxOnFileSendRequestCallback(private var ctx: Context) {

  def execute(friend: AntoxFriend,
    filenumber: Int,
    filesize: Long,
    filename: Array[Byte]) {
    Log.d(TAG, "execute")
    ToxSingleton.fileSendRequest(friend.getAddress, filenumber, new String(filename), filesize, ctx)
  }
}
